.github/workflows/interface_auto.yml

```diff
@@ -0,0 +1,72 @@
+name: Deployment on DockerHub and Hugging Face
+
+# Trigger the workflow on push events to the 'uncleaned-branch' branch
+on:
+  push:
+    branches:
+      - uncleaned-branch
```

Comment:
I would use the following instead so that the workflow gets triggered once a new pull request is made to the main branch.

```python
  push:
    branches:
      - "master"
      - "main"
  pull_request:
    branches:
      - "master"
      - "main"
```